runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1417,3 +1417,113 @@ TR_J9ServerVM::dereferenceStaticFinalAddress(void *staticAddress, TR::DataType a
    return it->second;
    }
 
+bool
+TR_J9SharedCacheServerVM::isClassLibraryMethod(TR_OpaqueMethodBlock *method, bool vettedForAOT)
+   {
+   TR_ASSERT(vettedForAOT, "The TR_J9SharedCacheServerVM version of this method is expected to be called only from isClassLibraryMethod.\n"
+                                          "Please consider whether you call is vetted for AOT!");
+
+   if (getSupportsRecognizedMethods())
+      return this->TR_J9ServerVM::isClassLibraryMethod(method, vettedForAOT);
+
+   return false;
+   }
+
+bool
+TR_J9SharedCacheServerVM::stackWalkerMaySkipFrames(TR_OpaqueMethodBlock *method, TR_OpaqueClassBlock *methodClass)
+   {
+   bool skipFrames = TR_J9ServerVM::stackWalkerMaySkipFrames(method, methodClass);
+   TR::Compilation *comp = TR::comp();
+   if (comp && comp->getOption(TR_UseSymbolValidationManager))
+      {
+      if (!comp->getSymbolValidationManager()->addStackWalkerMaySkipFramesRecord(method, methodClass, skipFrames))
+         {
+         TR_ASSERT(false, "Failed to validate addStackWalkerMaySkipFramesRecord\n");
+         }
+      }
+   return skipFrames;
+   }
+
+bool
+TR_J9SharedCacheServerVM::isMethodEnterTracingEnabled(TR_OpaqueMethodBlock *method)
+   {
+   // We want to return the same answer as TR_J9VMBase unless we want to force it to allow tracing
+   return TR_J9VMBase::isMethodEnterTracingEnabled(method) || TR::Options::getAOTCmdLineOptions()->getOption(TR_EnableAOTMethodEnter);
```

Comment:
This should ask `TR_J9ServerVM`

Reply:
The test on cmdLineOptions is not going to work here because those are common for all clients. We may need to fetch the comp object and find the options for this client

Reply:
Not able to find the TR_J9ServerVM::isMethodEnterTracingEnabled.
On client side also using the TR_J9VMBase::isMethodEnterTracingEnabled version not the TR_J9VM::isMethodEnterTracingEnabled.